### DISCUSSION TITLE
has'…' error

### DISCUSSION THREAD

**Discussion by xenoterracide** [z-shell/zi#194] [Q&A] [answered]
```zsh
zi wait has"brew" atload"$(brew --prefix asdf)/asdf.sh" for z-shell/null
/home/xeno/.zshrc:1: command not found: brew
```

shouldn't this not error? note: brew is not installed.

**Comment by ss-o** [comment]
It seems error comes from: `atload"$(brew --prefix asdf)/asdf.sh"`, I am not sure if it's correct. 

I will update you on this one.

**Comment by ss-o** [comment]
I have tried `zi wait has"brew" atload"$(brew --prefix asdf)/asdf.sh" for z-shell/null` and it worked for me.

I could check this if you provide steps to reproduce.

However, it seems brew is not in PATH. I would double-check PATH or try:

```zsh
zi wait has"brew" atload"$(<full/path/to/>brew --prefix asdf)/asdf.sh" for z-shell/null
```

**Reply by xenoterracide** [reply]
All I have to do is run it

```
❯ zi wait has'brew' atload"source $(brew --prefix asdf)/asdf.sh" for z-shell/nul
zsh: command not found: brew
````

I am sure that it's not on my path, because I don't install it on linux.

You can grab my dot files and install them with yadm and stick that command in the zshrc it created on linux. 

`PATH`
```
/home/xeno/.asdf/shims:/opt/asdf-vm/bin:/home/xeno/.zi/polaris/bin:/home/xeno/.local/bin:/home/xeno/.zi/plugins/z-shell---zconvey/cmds:/usr/local/bin:/usr/bin:/var/lib/snapd/snap/bin:/usr/local/sbin:/opt/android-sdk/cmdline-tools/latest/bin:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/home/xeno/.cargo/bin
```

are you sure you don't have brew on your path?

**Comment by ss-o** [comment, marked as answer]
It seems that `zsh: command not found: brew` is printed by `atload''`.

Solution:

call if exists cmd brew

```shell
zi wait has"brew" atload"source $($commands[brew] --prefix asdf)/asdf.sh" for z-shell/null
```

or (safer)

```shell
zi wait has"brew" atload"(( $+commands[brew] )) && source $(brew --prefix asdf)/asdf.sh" for z-shell/null
```

and probably (only if true: `has'brew'` and true: `file exists`)

```shell
zi wait has"brew" atload"[[ -f $(brew --prefix asdf)/asdf.sh ]] && source $(brew --prefix asdf)/asdf.sh" for z-shell/null
```

alternate syntax for above:

```shell
zi wait has"brew" atload"[[ ! -f $(brew --prefix asdf)/asdf.sh ]] || source $(brew --prefix asdf)/asdf.sh" for z-shell/null
```
